server/requestorserver/auth_test.go

```diff
@@ -170,3 +170,102 @@ func TestHmacAuthenticator_Authenticate(t *testing.T) {
+
+	requestHeaders := map[string][]string{
+		"Content-Type": {"text/plain"},
+	}
+
+	t.Run("valid", func(t *testing.T) {
+		applies, parsedRequest, requestor, err := authenticator.AuthenticateRevocation(requestHeaders, []byte(validJwtData))
+		if err != nil {
+			require.NoError(t, err)
+		}
+		require.True(t, applies)
+		require.Equal(t, "https://irma.app/ld/request/revocation/v1", parsedRequest.LDContext)
+		require.Equal(t, "bsn-123456789", parsedRequest.Key)
+		require.Equal(t, "my_requestor", requestor)
+	})
+
+	server.Logger.SetLevel(logrus.ErrorLevel)
+	t.Run("invalid jwt requestor", func(t *testing.T) {
+		j := NewRevocationJwt("another_requestor", revocationRequest)
+		invalidJwtData, jErr := j.Sign(jwt.SigningMethodHS256, key)
+		require.NoError(t, jErr)
+
+		applies, _, _, err := authenticator.AuthenticateRevocation(requestHeaders, []byte(invalidJwtData))
+		require.True(t, applies)
+		require.Error(t, err)
+	})
+
+	t.Run("empty jwt data", func(t *testing.T) {
+		claims := (*jwt.MapClaims)(&map[string]interface{}{
+			"iss":        "my_requestor",
+			"iat":        time.Now().Unix(),
+			"revrequest": map[string]interface{}{},
+		})
+		emptyJwtData, jErr := jwt.NewWithClaims(jwt.SigningMethodHS256, claims).SignedString(key)
+		require.NoError(t, jErr)
+		applies, _, _, err := authenticator.AuthenticateRevocation(requestHeaders, []byte(emptyJwtData))
+		require.True(t, applies)
+		require.Error(t, err)
+		require.Equal(t, string(server.ErrorInvalidRequest.Type), err.ErrorName)
+	})
+
+	t.Run("old jwt data", func(t *testing.T) {
+		j := NewRevocationJwt("my_requestor", revocationRequest)
+		j.IssuedAt = (irma.Timestamp)(time.Unix(0, 0))
+		invalidJwtData, jErr := j.Sign(jwt.SigningMethodHS256, key)
+		require.NoError(t, jErr)
+		applies, _, _, err := authenticator.AuthenticateRevocation(requestHeaders, []byte(invalidJwtData))
+		require.True(t, applies)
+		require.Error(t, err)
+		require.Equal(t, string(server.ErrorUnauthorized.Type), err.ErrorName)
+	})
+
+	t.Run("jwt data not yet valid", func(t *testing.T) {
+		j := NewRevocationJwt("my_requestor", revocationRequest)
+		j.IssuedAt = (irma.Timestamp)(time.Now().AddDate(1, 0, 0))
+		invalidJwtData, jErr := j.Sign(jwt.SigningMethodHS256, key)
+		require.NoError(t, jErr)
+		applies, _, _, err := authenticator.AuthenticateRevocation(requestHeaders, []byte(invalidJwtData))
+		require.True(t, applies)
+		require.Error(t, err)
+		require.Equal(t, string(server.ErrorInvalidRequest.Type), err.ErrorName)
+	})
+
+	t.Run("jwt signed using invalid key", func(t *testing.T) {
+		j := NewRevocationJwt("my_requestor", revocationRequest)
+		invalidJwtData, jErr := j.Sign(jwt.SigningMethodHS256, invalidKey)
+		require.NoError(t, jErr)
+		applies, _, _, err := authenticator.AuthenticateRevocation(requestHeaders, []byte(invalidJwtData))
+		require.True(t, applies)
+		require.Error(t, err)
+	})
+}
+
+func NewRevocationJwt(servername string, rr *irma.RevocationRequest) *irma.RevocationJwt {
```

Comment:
This function can be private if I'm right (`newRevocationJwt`).

Reply:
Done